### PR TITLE
UF-6158 Refaktorisera service.

### PR DESCRIPTION
**Pull request by AndreSoderlund**
Blev en viss scope creep i samband med refaktorisering av serviceklassen. Skrivit om stora delar av testerna, uppdaterat integrationen till messaging, gått från application.properties till application.yml

### CODE REVIEW

File: src/main/java/se/sundsvall/incident/api/IncidentResource.java [hunk 1]
```diff
@@ -41,165 +43,106 @@ public class IncidentResource {
...
-	@GetMapping("/setincidentfeedback")
-	ResponseEntity<Void> setIncidentFeedback(@RequestParam(name = "errandid") String incidentId, @RequestParam(name = "feedback") String feedback) {
+	@Operation(summary = "Set incident feedback",
+		responses = {
+			@ApiResponse(responseCode = "200", description = "Successful Operation", useReturnTypeSchema = true),
+			@ApiResponse(responseCode = "400", description = "Bad Request",
+				content = @Content(mediaType = APPLICATION_PROBLEM_JSON_VALUE, schema = @Schema(implementation = Problem.class))),
+			@ApiResponse(responseCode = "404", description = "Not Found",
+				content = @Content(mediaType = APPLICATION_PROBLEM_JSON_VALUE, schema = @Schema(implementation = Problem.class))),
+			@ApiResponse(responseCode = "500", description = "Internal Server Error",
```
Comment (cheezi0747) [inline src/main/java/se/sundsvall/incident/api/IncidentResource.java:134]: Move 400 and 500 responses to class level as you have them on every method

File: src/main/java/se/sundsvall/incident/api/ValidListResource.java [hunk 2]
```diff
@@ -86,11 +88,7 @@ public ResponseEntity<List<ValidCategoryResponse>> getValidCategories() {
 		content = @Content(schema = @Schema(implementation = Problem.class)))
```
Comment (cheezi0747) [inline src/main/java/se/sundsvall/incident/api/ValidListResource.java:88]: Move 400 and 500 responses to class level as you have them on every method

File: src/main/java/se/sundsvall/incident/api/model/IncidentResponse.java [hunk 3]
```diff
@@ -15,14 +18,14 @@ public class IncidentResponse {
 	private String incidentId;
```
Comment (cheezi0747) [inline src/main/java/se/sundsvall/incident/api/model/IncidentResponse.java:18]: All fields in this should have a schema description

File: src/main/java/se/sundsvall/incident/service/CategoryService.java [hunk 4]
```diff
@@ -5,64 +5,92 @@
...
 import se.sundsvall.incident.integration.db.entity.CategoryEntity;
+import se.sundsvall.incident.integration.db.repository.CategoryRepository;
 import se.sundsvall.incident.service.mapper.Mapper;
 
 @Service
 public class CategoryService {
-
-	private static final String CATEGORY_NOT_FOUND = "Category with id: %s not found";
+	private static final String ENTITY_NOT_FOUND = "%s with id: %s not found";
+	private static final String CATEGORY = "Category";
```
Comment (cheezi0747) [inline src/main/java/se/sundsvall/incident/service/CategoryService.java:26]: CATEGORY is only used in the ENTITY_NOT_FOUND string. Include it in the string right away?

File: src/test/java/se/sundsvall/incident/service/mapper/MapperTest.java [hunk 5]
```diff
@@ -3,23 +3,26 @@
...
+import static se.sundsvall.incident.service.mapper.Mapper.toIncidentEntity;
+import static se.sundsvall.incident.service.mapper.Mapper.toIncidentOepResponse;
+import static se.sundsvall.incident.service.mapper.Mapper.toIncidentResponse;
 
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
-import org.mockito.InjectMocks;
 import org.mockito.junit.jupiter.MockitoExtension;
 
 @ExtendWith(MockitoExtension.class)
```
Comment (foobar-1337) [inline src/test/java/se/sundsvall/incident/service/mapper/MapperTest.java:18]: Is this extension necessary here?

File: src/main/resources/db/migration/V2__update_tables.sql [hunk 6]
```diff
@@ -92,7 +84,14 @@ ALTER TABLE IF EXISTS attachment
     MODIFY file LONGTEXT NOT NULL;
 
 
+ALTER TABLE IF EXISTS incident
+    DROP COLUMN ID;
```
Comment (cheezi0747) [inline src/main/resources/db/migration/V2__update_tables.sql:88]: will cause issue if table exists and column id does not

File: src/main/resources/db/migration/V2__update_tables.sql [hunk 7]
```diff
@@ -74,9 +68,7 @@ ALTER TABLE IF EXISTS incident
     MODIFY updated DATETIME NOT NULL;
 
 ALTER TABLE IF EXISTS incident
-    MODIFY category ENUM('BELYSNING', 'BRO_TUNNEL_KONSTRUKTION', 'FELPARKERAD_BIL', 'GANGCYKELVAG', 'KLOTTER',
-    'PARK_SKOG_LEKPLATS', 'PARKERING', 'RENHALLNING', 'SKYLT_VAGMARKE', 'TRAFIKSIGNAL', 'VAG_GATA',
-    'VAGMARKERING', 'OVRIGT', 'LIVBOJ', 'LIVBAT', 'VATTENMATARE', 'BRADD_OVERVAKNINGS_LARM');
+    RENAME COLUMN Category TO category_id;
```
Comment (cheezi0747) [inline src/main/resources/db/migration/V2__update_tables.sql:71]: will cause issue if table exists and column does not

File: src/main/resources/db/migration/V2__update_tables.sql [hunk 8]
```diff
@@ -92,7 +84,14 @@ ALTER TABLE IF EXISTS attachment
     MODIFY file LONGTEXT NOT NULL;
 
 
+ALTER TABLE IF EXISTS incident
+    DROP COLUMN ID;
+
+ALTER TABLE IF EXISTS incident
+    RENAME COLUMN IncidentId TO id;
```
Comment (cheezi0747) [inline src/main/resources/db/migration/V2__update_tables.sql:91]: will cause issue if table exists and column does not

File: src/main/java/se/sundsvall/incident/service/CategoryService.java [hunk 9]
```diff
@@ -5,64 +5,92 @@
...
-		var patchedEntity = categoryRepository.save(patchEntity(categoryEntity, patch));
-		return toCategoryDto(patchedEntity);
+			.orElseThrow(() -> Problem.valueOf(NOT_FOUND, ENTITY_NOT_FOUND.formatted(CATEGORY, id)));
+		var category = categoryRepository.save(patchCategory(categoryEntity, patch));
+		return toCategoryDto(category);
+	}
+
+	public List<ValidCategoryResponse> fetchValidCategories() {
+		List<ValidCategoryResponse> validCategories = new ArrayList<>();
+		var categories = categoryRepository.findAll();
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/incident/service/CategoryService.java:66]: Could be simplified with something like this:

```
return categoryRepository.findAll().stream()
   .map(category -> ValidCategoryResponse.builder()
				.withCategory(category.getLabel())
				.withCategoryId(category.getCategoryId())
				.build()))
   .toList();
```

File: src/main/java/se/sundsvall/incident/service/CategoryService.java [hunk 10]
```diff
@@ -5,64 +5,92 @@
...
+				.withCategory(category.getLabel())
+				.withCategoryId(category.getCategoryId())
+				.build());
+		}
+		return validCategories;
 	}
 
-	CategoryEntity patchEntity(final CategoryEntity entity, final CategoryPatch patch) {
+	public List<ValidOepCategoryResponse> fetchValidOepCategories() {
+		List<ValidOepCategoryResponse> validOepCategories = new ArrayList<>();
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/incident/service/CategoryService.java:78]: See comment above

File: src/main/java/se/sundsvall/incident/service/IncidentService.java [hunk 11]
```diff
@@ -1,105 +1,130 @@
...
+		try {
+			switch (incident.getCategory().getTitle()) {
+				case "LIVBAT", "LIVBOJ" ->
+					incident.setExternalCaseId(lifeBuoyIntegration.sendLifeBuoy(incident));
+				case "VATTENMATARE", "BRADD_OVERVAKNINGS_LARM" ->
+					messagingIntegration.sendMSVAEmail(incident);
+				default -> messagingIntegration.sendEmail(incident);
+			}
+		} catch (Exception e) {
+			incident.setStatus(Status.ERROR);
```
Comment (cheezi0747) [inline src/main/java/se/sundsvall/incident/service/IncidentService.java:125]: this status will never be saved

File: src/main/resources/db/migration/V2__update_tables.sql (hunk 6 above)
Comment (AndreSoderlund) [reply src/main/resources/db/migration/V2__update_tables.sql:88]: ```ALTER TABLE IF EXISTS incident DROP COLUMN IF EXISTS ID;```

Would this resolve the issue?

File: src/test/java/se/sundsvall/incident/integration/messaging/MessagingMapperTest.java [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.thymeleaf.spring6.SpringTemplateEngine;
+
+import se.sundsvall.incident.integration.messaging.configuration.MessagingProperties;
+
+@ExtendWith(MockitoExtension.class)
+class MessagingMapperTest {
+
+	@Mock
+	private MessagingProperties mockProperties;
```
Comment (josc04) [inline src/test/java/se/sundsvall/incident/integration/messaging/MessagingMapperTest.java:25]: Prefix or suffix with Mock?

File: src/test/java/se/sundsvall/incident/integration/messaging/MessagingMapperTest.java [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+
+	@Mock
+	private SpringTemplateEngine thymeleafTemplateEngine;
```
Comment (josc04) [inline src/test/java/se/sundsvall/incident/integration/messaging/MessagingMapperTest.java:28]: Prefix or suffix with Mock?

File: src/test/java/se/sundsvall/incident/integration/messaging/MessagingMapperTest.java [hunk 14]
```diff
@@ -0,0 +1,111 @@
...
+
+	private MessagingMapper messagingMapper;
```
Comment (josc04) [inline src/test/java/se/sundsvall/incident/integration/messaging/MessagingMapperTest.java:30]: Won't `@InjectMocks` work here (and remove of setup method)?

File: src/test/java/se/sundsvall/incident/integration/messaging/MessagingMapperTest.java [hunk 15]
```diff
@@ -0,0 +1,111 @@
...
+		var decodedMessage = Base64.getDecoder().decode(response.getHtmlMessage());
+		var email = new String(decodedMessage, StandardCharsets.UTF_8);
+		assertThat(email).contains("Bräddövervakningslarm");
+		assertThat(response).isNotNull();
+		assertThat(response.getSubject()).isEqualTo("this is a subject asdas - asdasd");
+		assertThat(response.getEmailAddress()).isEqualTo("nowhere@nowhere.com");
+		assertThat(response.getSender().getName()).isEqualTo("someEmailName");
+		assertThat(response.getSender().getAddress()).isEqualTo("someemail@host.se");
+		assertThat(response.getSender().getReplyTo()).isEqualTo("elsewhere@email.com");
+	}
```
Comment (josc04) [inline src/test/java/se/sundsvall/incident/integration/messaging/MessagingMapperTest.java:59]: Add verifications that mocks has been called?

File: src/test/java/se/sundsvall/incident/integration/messaging/MessagingMapperTest.java [hunk 16]
```diff
@@ -0,0 +1,111 @@
...
+		var decodedMessage = Base64.getDecoder().decode(response.getHtmlMessage());
+		var email = new String(decodedMessage, StandardCharsets.UTF_8);
+		assertThat(email).contains("Vattenmätare");
+		assertThat(response).isNotNull();
+		assertThat(response.getSubject()).isEqualTo("this is a subject asdas - asdasd");
+		assertThat(response.getEmailAddress()).isEqualTo("nowhere@nowhere.com");
+		assertThat(response.getSender().getName()).isEqualTo("someEmailName");
+		assertThat(response.getSender().getAddress()).isEqualTo("someemail@host.se");
+		assertThat(response.getSender().getReplyTo()).isEqualTo("elsewhere@email.com");
+	}
```
Comment (josc04) [inline src/test/java/se/sundsvall/incident/integration/messaging/MessagingMapperTest.java:83]: Add verifications that mocks has been called?

File: src/test/java/se/sundsvall/incident/integration/messaging/MessagingMapperTest.java [hunk 17]
```diff
@@ -0,0 +1,111 @@
...
+
+	@Test
+	void toMSVAEmailRequest_Vattenmatare() {
```
Comment (josc04) [inline src/test/java/se/sundsvall/incident/integration/messaging/MessagingMapperTest.java:62]: Is there any significant difference between this test and toMSVAEmailRequest_Braddovervakningslarm (other than category) or could these two tests be parameterized?

File: src/test/java/se/sundsvall/incident/integration/messaging/MessagingMapperTest.java [hunk 18]
```diff
@@ -0,0 +1,111 @@
...
+
+		when(mockProperties.sender()).thenReturn(sender);
+		final var response = messagingMapper.toEmailDto(incident);
+
+		assertThat(response).isNotNull();
+		assertThat(response.getEmailAddress()).isEqualTo(incident.getCategory().getForwardTo());
+		assertThat(response.getSender().getName()).isEqualTo("someEmailName");
+		assertThat(response.getSender().getAddress()).isEqualTo("someemail@host.se");
+		assertThat(response.getAttachments()).hasSize(2);
+	}
```
Comment (josc04) [inline src/test/java/se/sundsvall/incident/integration/messaging/MessagingMapperTest.java:101]: Add verifications that mocks has been called?

File: src/test/java/se/sundsvall/incident/service/CategoryServiceTest.java [hunk 19]
```diff
@@ -37,28 +40,24 @@ class CategoryServiceTest {
...
-		assertThatThrownBy(() -> categoryService.fetchCategoryById(id))
-			.hasMessageContaining("Category with id: " + id + " not found")
-			.isInstanceOf(Problem.class);
-
-		verify(mockCategoryRepository, times(1)).findById(any());
-		verifyNoMoreInteractions(mockCategoryRepository);
+		assertThatThrownBy(() -> categoryService.fetchCategoryById(any(Integer.class)))
+			.isInstanceOf(Problem.class)
+			.hasMessageContaining("Not Found: Category");
 	}
```
Comment (josc04) [inline src/test/java/se/sundsvall/incident/service/CategoryServiceTest.java:61]: Why not keep verifications towards mocks?

File: src/test/java/se/sundsvall/incident/service/CategoryServiceTest.java [hunk 20]
```diff
@@ -37,28 +40,24 @@ class CategoryServiceTest {
...
 		when(mockCategoryRepository.findById(any(Integer.class)))
-			.thenReturn(Optional.ofNullable(createCategoryEntity()));
+			.thenReturn(Optional.of(createCategoryEntity()));
 
 		var result = categoryService.fetchCategoryById(5);
 
-		assertThat(result).isInstanceOf(CategoryDTO.class).isNotNull();
+		assertThat(result).isInstanceOf(Category.class).isNotNull();
 
 		verify(mockCategoryRepository, times(1)).findById(any());
```
Comment (josc04) [inline src/test/java/se/sundsvall/incident/service/CategoryServiceTest.java:49]: `times(1)` is redundant (comment goes for all tests in class)

File: src/test/java/se/sundsvall/incident/service/CategoryServiceTest.java [hunk 21]
```diff
@@ -101,73 +100,70 @@ void createCategoryTest() {
...
 
 	@Test
-	void deleteCategoryByIdWhenFoundTest() {
-		when(mockCategoryRepository.existsById(any(Integer.class))).thenReturn(true);
+	void deleteCategoryByIdTest() {
+		when(mockCategoryRepository.existsById(5)).thenReturn(true);
 
 		categoryService.deleteCategoryById(5);
-
 		verify(mockCategoryRepository, times(1)).existsById(any());
```
Comment (josc04) [inline src/test/java/se/sundsvall/incident/service/CategoryServiceTest.java:107]: Why not verify the actual parameter value (5)?

File: src/test/java/se/sundsvall/incident/service/CategoryServiceTest.java [hunk 22]
```diff
@@ -101,73 +100,70 @@ void createCategoryTest() {
...
+		when(mockCategoryRepository.existsById(anyInt())).thenReturn(false);
 
-		assertThatThrownBy(() -> categoryService.deleteCategoryById(id))
-			.hasMessageContaining("Category with id: " + id + " not found")
-			.isInstanceOf(Problem.class);
+		assertThatThrownBy(() -> categoryService.deleteCategoryById(anyInt()))
+			.isInstanceOf(Problem.class)
+			.hasMessageContaining("Not Found: Category");
 
-		verify(mockCategoryRepository, times(1)).existsById(any());
```
Comment (josc04) [inline src/test/java/se/sundsvall/incident/service/CategoryServiceTest.java:122]: Keep verfication but change to verify with actual value?

File: src/test/java/se/sundsvall/incident/service/CategoryServiceTest.java [hunk 23]
```diff
@@ -101,73 +100,70 @@ void createCategoryTest() {
...
-		verifyNoMoreInteractions(mockCategoryRepository);
+	@Test
+	void patchEntityNotFoundTest() {
+		var patch = createCategoryPatch();
+		when(mockCategoryRepository.findById(anyInt())).thenReturn(Optional.empty());
+
+		assertThatThrownBy(() -> categoryService.patchCategory(anyInt(), patch))
+			.isInstanceOf(Problem.class)
+			.hasMessageContaining("Not Found: Category");
 	}
```
Comment (josc04) [inline src/test/java/se/sundsvall/incident/service/CategoryServiceTest.java:143]: Verify the following:
```
verify(mockCategoryRepository).findById(anyInt());
verify(mockCategoryRepository, never()).save(any());
```

File: src/test/java/se/sundsvall/incident/integration/messaging/MessagingIntegrationTest.java [hunk 24]
```diff
@@ -15,36 +15,39 @@
...
 
 import generated.se.sundsvall.messaging.EmailRequest;
+import generated.se.sundsvall.messaging.MessageResult;
 
 @ExtendWith(MockitoExtension.class)
 class MessagingIntegrationTest {
 
 	@Mock
-	private EmailMapper emailMapper;
+	private MessagingMapper messagingMapper;
```
Comment (josc04) [inline src/test/java/se/sundsvall/incident/integration/messaging/MessagingIntegrationTest.java:24]: Prefix or suffix with Mock?

File: src/test/java/se/sundsvall/incident/service/IncidentServiceTest.java [hunk 25]
```diff
@@ -26,173 +27,209 @@
...
-	void getOepIncidentStatus() {
-		final var incidentEntity = buildIncidentEntity(Category.VATTENMATARE);
-		when(incidentRepository.findIncidentEntityByExternalCaseId(anyString())).thenReturn(Optional.of(incidentEntity));
+	void fetchIncidentByIdTest() {
+		when(mockIncidentRepository.findById(anyString())).thenReturn(Optional.ofNullable(createIncidentEntity()));
+
+		var result = incidentService.fetchIncidentById(anyString());
+
+		assertThat(result).isNotNull().isInstanceOf(IncidentResponse.class);
+		verify(mockIncidentRepository, times(1)).findById(anyString());
```
Comment (josc04) [inline src/test/java/se/sundsvall/incident/service/IncidentServiceTest.java:69]: `times(1)` is redunant and can be removed

(Comment goes for all rows in class where `times(1)` is used)

File: src/test/java/se/sundsvall/incident/service/IncidentServiceTest.java [hunk 26]
```diff
@@ -26,173 +27,209 @@
...
-		final var response = incidentService.getOepIncidentStatus("123").orElse(null);
+	@Test
+	void fetchIncidentById_NotFoundTest() {
+		when(mockIncidentRepository.findById(anyString())).thenReturn(Optional.empty());
 
-		assertThat(response).isNotNull();
-		assertThat(response.getIncidentId()).isEqualTo(incidentEntity.getIncidentId());
+		assertThatThrownBy(() -> incidentService.fetchIncidentById(anyString()))
+			.hasMessageContaining("Not Found: Incident with id:  not found")
+			.isInstanceOf(Problem.class);
```
Comment (josc04) [inline src/test/java/se/sundsvall/incident/service/IncidentServiceTest.java:80]: Add verification of interaction with mock?

File: src/test/java/se/sundsvall/incident/service/IncidentServiceTest.java [hunk 27]
```diff
@@ -26,173 +27,209 @@
...
-		assertThat(response.getStatusText()).isEqualTo(Status.INSKICKAT.getLabel());
-		verify(incidentRepository, times(1)).save(any());
-		verify(messagingIntegration, times(1)).sendMSVAEmail(any(IncidentDto.class));
+	void fetchOepIncidentStatus_NotFoundTest() {
+		when(mockIncidentRepository.findIncidentEntityByExternalCaseId(anyString())).thenReturn(Optional.empty());
+
+		assertThatThrownBy(() -> incidentService.fetchOepIncidentStatus(anyString()))
+			.hasMessageContaining("Not Found: Incident with id:  not found")
+			.isInstanceOf(Problem.class);
 	}
```
Comment (josc04) [inline src/test/java/se/sundsvall/incident/service/IncidentServiceTest.java:102]: Add verification of interaction with mock?

File: src/test/java/se/sundsvall/incident/service/IncidentServiceTest.java [hunk 28]
```diff
@@ -26,173 +27,209 @@
...
-		final var incidentEntity = buildIncidentEntity(Category.KLOTTER);
-		when(incidentRepository.save(any())).thenReturn(incidentEntity);
-		doThrow(new NullPointerException()).when(messagingIntegration).sendEmail(any(IncidentDto.class));
-
-		final var response = incidentService.sendIncident(incidentSaveRequest);
-		assertThat(response).isNotNull();
-		assertThat(response.getStatusText()).isEqualTo(Status.ERROR.getLabel());
-		verify(incidentRepository, times(1)).save(any());
-		verify(messagingIntegration, times(1)).sendEmail(any(IncidentDto.class));
+	void fetchPaginatedIncidentsTest() {
```
Comment (josc04) [inline src/test/java/se/sundsvall/incident/service/IncidentServiceTest.java:105]: Add test with for Optional.empty() to verify default values for page and pageSize?

File: src/test/java/se/sundsvall/incident/service/IncidentServiceTest.java [hunk 29]
```diff
@@ -26,173 +27,209 @@
...
+		var request = createIncidentSaveRequest();
+		when(mockCategoryRepository.findById(anyInt())).thenReturn(Optional.empty());
 
-		verify(incidentRepository, times(1)).findById(anyString());
-		verify(attachmentRepository, times(1)).findAllByIncidentId(anyString());
+		assertThatThrownBy(() -> incidentService.createIncident(request))
+			.isInstanceOf(Problem.class)
+			.hasMessageContaining("Not Found: Category with id: ");
 
+		verify(mockIncidentRepository, never()).save(any());
```
Comment (josc04) [inline src/test/java/se/sundsvall/incident/service/IncidentServiceTest.java:139]: Keep verification that findById is called?

File: src/integration-test/java/apptest/MessagingIT.java [hunk 30]
```diff
@@ -1,97 +1,66 @@
...
 
-	@Autowired
-	private IncidentRepository incidentRepository;
+	private static final String PATH = "/incident";
+	private static final String REQUEST_FILE = "request.json";
+	private static final String EXPECTED_FILE = "expected.json";
 
 	@BeforeEach
-	void setUp() {
+	void setup() {
```
Comment (cheezi0747) [inline src/integration-test/java/apptest/MessagingIT.java:29]: should be setUp probably. more correct I think?

File: src/test/java/se/sundsvall/incident/service/IncidentServiceTest.java [hunk 31]
```diff
@@ -26,173 +27,209 @@
...
+		category.setTitle("VATTENMATARE");
+		entity.setCategory(category);
 
-		incidentService.updateIncidentFeedback(INCIDENTID, "someFeedback");
-		verify(incidentRepository, times(1)).findById(any());
-		verify(incidentRepository, times(1)).save(any());
+		incidentService.sendNotification(entity);
+
+		verify(mockMessagingIntegration, times(1)).sendMSVAEmail(any());
 	}
```
Comment (josc04) [inline src/test/java/se/sundsvall/incident/service/IncidentServiceTest.java:220]: Add test for sending notification for other than vattenmätare and livbåt to verify default behavior in service (using messagingIntegration.sendEmail)

File: src/test/java/se/sundsvall/incident/service/mapper/MapperTest.java [hunk 32]
```diff
@@ -3,23 +3,23 @@
...
 
-	@InjectMocks
-	private Mapper mapper;
-
 	@Test
 	void toCategoryDtoTest() {
 		var entity = createCategoryEntity();
 
-		var dto = mapper.toCategoryDto(entity);
+		var dto = toCategoryDto(entity);
```
Comment (josc04) [inline src/test/java/se/sundsvall/incident/service/mapper/MapperTest.java:22]: My OCD kicking in, but change dto to category?

File: src/test/java/se/sundsvall/incident/service/mapper/MapperTest.java [hunk 33]
```diff
@@ -47,8 +47,37 @@ void toCategoryEntityTest() {
...
+
+	@Test
+	void toIncidentEntityWhenNullTest() {
+		var entity = toIncidentEntity(null, null, null);
+		assertThat(entity).isNull();
+	}
+
+	@Test
+	void toIncidentResponseWhenNullTest() {
+		var entity = toIncidentResponse(null);
```
Comment (josc04) [inline src/test/java/se/sundsvall/incident/service/mapper/MapperTest.java:62]: Change var name as it is not an entity

File: src/test/java/se/sundsvall/incident/service/mapper/MapperTest.java [hunk 34]
```diff
@@ -47,8 +47,37 @@ void toCategoryEntityTest() {
...
+		assertThat(entity).isNull();
+	}
+
+	@Test
+	void toIncidentOepResponseWhenNullTest() {
+		var entity = toIncidentOepResponse(null);
```
Comment (josc04) [inline src/test/java/se/sundsvall/incident/service/mapper/MapperTest.java:68]: Change var name as it is not an entity

File: src/test/java/se/sundsvall/incident/service/mapper/MapperTest.java [hunk 35]
```diff
@@ -47,8 +47,37 @@ void toCategoryEntityTest() {
...
+		assertThat(entity).isNull();
+	}
 
+	@Test
+	void toAttachmentWhenNullTest() {
+		var entity = toAttachment(null);
```
Comment (josc04) [inline src/test/java/se/sundsvall/incident/service/mapper/MapperTest.java:74]: Change var name as it is not an entity

File: src/test/java/se/sundsvall/incident/api/CategoryResourceTest.java [hunk 36]
```diff
@@ -73,4 +79,32 @@ void deleteCategoryByIdTest_shouldReturn204_whenDeleted() {
...
 
+	@Test
+	void getValidCategories() {
+		when(mockCategoryService.fetchValidCategories()).thenReturn(List.of(createValidCategoryResponse()));
+
+		var response = categoryResource.getValidCategories();
+		assertThat(response).isNotNull();
+		assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+		assertThat(response.getBody()).isNotNull();
+		assertThat(response.getBody()).hasSize(1);
```
Comment (cheezi0747) [inline src/test/java/se/sundsvall/incident/api/CategoryResourceTest.java:90]: Du kan oneline det här. (Gäller metoden under denna också)

`assertThat(response.getBody()).isNotNull().hasSize(1);`

**Review by josc04 — APPROVED** [review]

**Pull request merged**